src/Service/ProducerEventProcessor.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SERVICE_PRODUCER_EVENT_PROCESSOR_H_
+#define SERVICE_PRODUCER_EVENT_PROCESSOR_H_
+
+#include <stdint.h>
+
+#include "CaptureEventBuffer.h"
+#include "capture.pb.h"
+
+namespace orbit_service {
+//
+// The implementation of this interface is responsible
+// for processing ProducerCaptureEvents from multiple producers
+// It converts them to ClientCaptureEvents and sends to CaptureEventBuffer
+//
+class ProducerEventProcessor {
+ public:
+  ProducerEventProcessor() = default;
+  virtual ~ProducerEventProcessor() = default;
+
+  virtual void ProcessEvent(uint64_t producer_id,
+                            orbit_grpc_protos::ProducerCaptureEvent event) = 0;
+};
+
+std::unique_ptr<ProducerEventProcessor> CreateProducerEventProcessor(
+    CaptureEventBuffer* capture_event_buffer);
```

Comment:
By the way, this seems like a very Java-like pattern to me, were all object variables are references. Why can't `ProducerEventProcessorImpl` be in its own file and be instantiated directly, instead of having to always be wrapped in a `unique_ptr`?

Reply:
Yeah, this is certainly a pro, what I don't like of the pattern is that I'm forcibly given a pointer and for example the class can never be stack allocated. So pros and cons, I'm fine with it.